server/src/main/java/org/elasticsearch/index/shard/IndexShard.java

```diff
@@ -1235,87 +1235,7 @@ private void markSearcherAccessed() {
         readAllowed();
         markSearcherAccessed();
         final Engine engine = getEngine();
-        return engine.acquireSearcher(source, scope, this::wrapSearcher);
-    }
-
-    private Engine.Searcher wrapSearcher(Engine.Searcher searcher) {
-        assert ElasticsearchDirectoryReader.unwrap(searcher.getDirectoryReader())
-            != null : "DirectoryReader must be an instance or ElasticsearchDirectoryReader";
-        boolean success = false;
-        try {
-            final Engine.Searcher newSearcher = readerWrapper == null ? searcher : wrapSearcher(searcher, readerWrapper);
-            assert newSearcher != null;
-            success = true;
-            return newSearcher;
-        } catch (IOException ex) {
-            throw new ElasticsearchException("failed to wrap searcher", ex);
-        } finally {
-            if (success == false) {
-                Releasables.close(success, searcher);
-            }
-        }
-    }
-
-    static Engine.Searcher wrapSearcher(Engine.Searcher engineSearcher,
```

Comment:
I think we could leave this method where it is, since it is only used by IndexShard (except for tests)?

It might require a hack in the test code though to not have it public (similar to EngineAccess), but my preference would be to do that. But is kind of a small thing, can also accept this as is.

Reply:
Yeah, I reverted it in https://github.com/elastic/elasticsearch/pull/64504/commits/a1dcfed273260b96604b799c2c624f3823ed15b0.